webapp/app/plugins/pinboard.js

```diff
@@ -108,12 +108,28 @@ treeherder.controller('PinboardCtrl', [
         $scope.toggleEnterBugNumber = function(tf) {
             $scope.enteringBugNumber = tf;
             $scope.focusInput = tf;
+            if (tf) {
+                // We want escape to first cancel entering the bug number, so
+                // unbind the hotkey. Pressing escape again will close the
+                // panels as usual.
+                Mousetrap.unbind(['escape']);
+            } else {
+                $scope.newEnteredBugNumber = '';
+            }
         };
 
         $scope.completeClassification = function() {
             $rootScope.$broadcast('blur-this', "classification-comment");
         };
 
+        $scope.checkCancelEnterBugNumber = function(event) {
+            if (event.keyCode == 27 /* DOM_VK_ESCAPE */ || event.key == 'r') {
```

Comment:
Taken care of, it seems rebinding with Mousetrap works without issues. I checked with a debugger breakpoint to make sure its not still bound to closing the bug field after the pinboard is closed.